internal/js_library/js_library.bzl

```diff
@@ -32,10 +32,12 @@ load(
 )
 
 _ATTRS = {
+    "ambient_srcs": attr.label_list(allow_files = True),
```

Comment:
your feature is for `ts_library` but not needed for `ts_project`. That makes it feel undesirable to me for the layering to require that js_library know about this stuff.
Imagine I'm just following the TypeScript documentation. This wouldn't make any sense to me.

What would this PR look like if you didn't change `js_library`? Maybe it shakes out one of these ways:
1. there's some example you can't do like provide a `.js`/`.d.ts` pair exporting an ambient, but we could add that in a followup (just to defer this discussion and make it easier to land a first PR). But since `ts_library` already has a `module_name` field I don't see why `js_library` would be needed in such a case
2. these two new field could just be tucked into DeclarationInfo (but I guess the srcs one couldn't)
3. users have to put a `ts_library` as the srcs of a `js_library` to make this work

Reply:
after reading the rest, I think maybe it's 4) we need this just for third-party libraries but there isn't a usecase for first-party so we should document these new attrs as internal use only, not meant as public API

Reply:
If someone is trying to properly describe a `.d.ts` file using `js_library` they would need to use these attributes.
I think it is more common for people to write code as modules, but I don't think this is quite internal only.